extensions/authentication-passport/src/__tests__/acceptance/authentication-with-passport-strategy-adapter.acceptance.ts

```diff
@@ -29,62 +31,154 @@ import {
 } from '@loopback/rest';
 import {SecurityBindings, securityId, UserProfile} from '@loopback/security';
 import {Client, createClientForHandler} from '@loopback/testlab';
-import {BasicStrategy} from 'passport-http';
+import {BasicStrategy, BasicVerifyFunction} from 'passport-http';
 import {StrategyAdapter} from '../../strategy-adapter';
+
 const SequenceActions = RestBindings.SequenceActions;
 const AUTH_STRATEGY_NAME = 'basic';
+const INVALID_USER_CREDENTIALS_MESSAGE = 'Invalid credentials';
+const INVALID_USER_CREDENTIALS_CODE = 'INVALID_USER_CREDENTIALS';
+const USERS_REPOSITORY_BINDING_KEY = 'repositories.users';
+const VERIFY_FUNCTION_BASIC_AUTHENTICATION_BINDING_KEY =
+  'authentication.basic.verify';
+const BASIC_STRATEGY_BINDING_KEY =
+  'authentication.strategies.basicAuthStrategy';
+
+enum ScenarioEnum {
+  WithoutProviders,
+  WithProviders,
+}
 
 describe('Basic Authentication', () => {
   let app: Application;
   let server: RestServer;
   let users: UserRepository;
-  beforeEach(givenAServer);
+
+  //these setup steps are the same regardless of which scenario ( with or without providers)
+  beforeEach(givenAnApp);
   beforeEach(givenUserRepository);
   beforeEach(givenControllerInApp);
-  beforeEach(givenAuthenticatedSequence);
 
   it('authenticates successfully for correct credentials', async () => {
+    await remainingSetup(ScenarioEnum.WithoutProviders);
     const client = whenIMakeRequestTo(server);
     const credential =
-      users.list.joe.profile[securityId] + ':' + users.list.joe.password;
+      users.list.Joseph.profile.username + ':' + users.list.Joseph.password;
```

Comment:
Hmm, why encoding `username` instead of `[securityId]` here? Usually it's hard to make sure the combination of "username" and "password" is unique.

Reply:
`username` is not the same as `name` (which is a concatenation of `firstName` and `lastName`).
It is like `emonddr` username that I have for github...it is unique. Just like an email address is unique.

```
Joseph: {
        profile: {
          id: '1',
          username: 'joesmith71',
          firstName: 'Joseph',
          lastName: 'Smith',
        },
        password: '12345',
      }
```

I will add `71` to my username fields in my test cases to make this more clear. ;)